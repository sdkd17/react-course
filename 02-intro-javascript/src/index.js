

const apiKey = 'nJgFoWKmfae1ZgLV4aOMdPlNp4VsVJVS';
const tag = 'NBA';
const rating = 'g';

const peticion = fetch(`https://api.giphy.com/v1/gifs/random?api_key=${apiKey}&tag=${tag}&rating=${rating}`);

// peticion.then ( (resp) => {
//   resp.json().then( ( data ) => {
//     console.log(data);
//   })
// })
// .catch( console.wanr );

peticion
  .then ( resp => resp.json())
  .then ( ({ data } ) => {
    console.log(data);
    const { url } = data.images.original;
    console.log(url)

    const img = document.createElement('img');
    img.src = url;


    document.body.append(img);
  })
.catch( console.wanr );

