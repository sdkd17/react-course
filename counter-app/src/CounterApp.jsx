import PropType from 'prop-types';
import { useState } from 'react';



export const CounterApp = ( {value} ) => {

  const [counter, setCounter] = useState( value );

  const handleAdd = () => {
    setCounter((c) => c+1 );
  }
  
  const handleSub = () => {
    setCounter((c) => c > 0 ? c = c-1 : c );
  }
  
  const handleReset = () => {
    setCounter((c) => value );
  }
 

  return (
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>

      <button onClick={ handleAdd }>+1</button>
      <button onClick={ handleSub }>-1</button>
      <button aria-label='btn-reset' onClick={ handleReset }>reset</button>
      
    </>
  )
}

CounterApp.propTypes = {
  value: PropType.number.isRequired,
}


