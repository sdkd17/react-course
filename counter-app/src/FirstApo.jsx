import PropTypes from 'prop-types';

export const FirstApo = ({title, subTitle}) => {
  return (
    <>
      <h1>{ title }</h1>

      <h5> { subTitle } </h5>
    </>
  )
}

FirstApo.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}
