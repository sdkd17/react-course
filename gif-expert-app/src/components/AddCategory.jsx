import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({target}) => {
    setInputValue( target.value );
  }

  const onSubmit = (event) => {
    event.preventDefault();

    const newInputValue = inputValue.trim().toLowerCase();
    if( newInputValue.length <= 1 ) return;

    onNewCategory(newInputValue);
    setInputValue('');
    //inputValue = ''; esta mal porque no renderiza nuevamente
  }

  return (
    <form onSubmit={ (event) => onSubmit(event) }>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={ inputValue }
        onChange={ onInputChange } //evento es un parametro implicito
      />
    </form>
  )
}

AddCategory.propTypes = {
  // categories: PropTypes.array,
  onNewCategory: PropTypes.func,
}