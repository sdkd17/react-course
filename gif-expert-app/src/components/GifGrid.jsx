import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useGetGifs } from '../hooks/useGetGifs';

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useGetGifs(category);

  return (
    <>

    
      <h3>{ category }</h3>
      {isLoading && ( <h2>Cargando...</h2> )}
      <div className='card-grid'>
        {/*
          images.map( img => (
            <li key={img.id}>{ img.title }</li>
            )
          )
          Utilizar desestructuracion */
          images.map( (img) => (
              <GifGridItem 
                key={ img.id }
                {...img} //Pasaje de parametros individuales
              />
            )
          )
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
