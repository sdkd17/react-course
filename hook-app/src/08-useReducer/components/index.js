
export * from './TodoList';
export * from './TodoAdd';