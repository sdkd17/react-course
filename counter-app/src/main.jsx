import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApo } from './FirstApo';

import './styles.css'


ReactDOM.createRoot( document.getElementById( 'root' ) ). render (
  <React.StrictMode>
    {/* <FirstApo title={"Titulo"} subTitle={"subtitulo"}/> */}
    <CounterApp value={100}/>
  </React.StrictMode>
)