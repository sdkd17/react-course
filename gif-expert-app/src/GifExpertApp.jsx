import { useState } from "react";
import { AddCategory, GifGrid } from "./components";


export const GifExpertApp = () => {

	const [categories, setCategories] = useState( [] );

	const onAddCategory = (newCategory) => {
		if( categories.includes(newCategory) ) return;
		
		setCategories( [...categories, newCategory] );
		// setCategories( cat => [...cat, 'newCat'] );
	}

	return (
		<>
			<h1>GifExpertApp</h1>

			<AddCategory 
				// setCategories={ setCategories }
				onNewCategory={ onAddCategory }
			/>
			
      { 
        categories.map( category => 
          ( <GifGrid 
              category={ category }
              key={ category }
            />
          )
        )
      }
			
		</>
	)
}
