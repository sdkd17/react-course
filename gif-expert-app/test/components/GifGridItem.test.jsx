import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components';

describe('Pruebas en GifGridItem', () => { 

  const title = 'title';
  const url = 'www.url.com';

  test('should match snapshot', () => { 

    const { container } = render( <GifGridItem title={ title } url={ url } /> );
    expect( container ).toMatchSnapshot();
  })

  // test('should not render if properties are not set', () => { 
  //   render( <GifGridItem titule={ title } /> );
    
  // })

  test('should show image and ALT correctly', () => { 

    render( <GifGridItem title={ title } url={ url } />);
    // expect( screen.getByRole('img').getAttribute('src')).toBe( url );
    // expect( screen.getByRole('img').getAttribute('alt') ).toBe( title );

    const {src, alt} = screen.getByRole('img');
    expect( src ).toContain( url );
    expect( alt ).toBe( title );
  })

  test('should show title inside component', () => { 

    render( <GifGridItem title={ title } url={ url } />);
    expect( screen.getByText( title ) ).toBeTruthy();
  })
})

