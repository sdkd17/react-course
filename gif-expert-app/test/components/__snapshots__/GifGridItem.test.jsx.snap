// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="www.url.com"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
